.github/workflows/gke.yaml

```diff
@@ -82,10 +82,12 @@ jobs:
           gcloud container clusters create ${{ env.clusterName }} \
             --labels "usage=pr,owner=${{ steps.vars.outputs.owner }}" \
             --zone ${{ env.zone }} \
-            --preemptible \
-            --image-type COS \
+            --image-type COS_CONTAINERD \
```

Comment:
The documentation refers that COS is being deprecated in clusters <1.19. To guarantee consistency for the test runs, we should pin the channel in here as well as the k8s version. For example:
```
--release-channel=regular
--cluster-version=1.19.8-gke.1600
```
This information can be retrieved with:
```
$ gcloud container get-server-config --zone us-west2-a
```

Reply:
I'm not sure what you're referring to 🤔

As mentioned in my commit message, my reading of the documentation is that `COS` is deprecated for >1.19, hence why we should switch to `COS_CONTAINERD`:

> Warning: Since GKE node version 1.19, the Container-Optimized OS with Docker (cos) variant is deprecated. Please migrate to the Container-Optimized OS with Containerd (cos_containerd) variant, which is now the default GKE node image.

Reply:
>As mentioned in my commit message, my reading of the documentation is that COS is deprecated for >1.19, hence why we should switch to COS_CONTAINERD:

I understand that but by not pinning the k8s version it means that we will select the default k8s version provided by the default release channel which is 1.18. That's why I suggested to pin k8s version and the release-channel.
![image](https://user-images.githubusercontent.com/5714066/114948789-07c73480-9e50-11eb-93f2-d2bfd51b6984.png)

Reply:
@aanm unfortunately there is not reliable way to pin k8s version in a release channel, because versions in release channels are being retired all the time. If we pin now, we will have trouble provisioning clusters in a month or two and will need to update this workflow after we get some failing builds.

Reply:
@nebril would that be possible? I just want to avoid being caught in surprise. Also it would be a precise way of coordinating which k8s versions are being tested by just looking at the table of truth.

Reply:
It can be done with but not "out of the box": `--cluster-version` does not allow anything other than the specific values coming from `gcloud container get-server-config`.

If we do, it will break less often than pinning a specific version, but it will still break every 2 to 3 months (whenever minor versions rotate) unless we proactively update workflows ourselves before the older versions get deprecated. As the [GKE docs](https://cloud.google.com/kubernetes-engine/versioning) state:

> GKE clusters support running Kubernetes versions from any supported minor release. At least two, if not three, minor versions are available at any given time. However, subsequent intermediate releases might change what versions are available for new clusters.

I agree that controlling the versions is comfortable, but I also think the argument can be made in the other direction: letting GKE/AKS/EKS dictate the K8s version ensures that Cilium always runs against a version that is officially supported by the provider, and should also be the one that clients use right now (or will use soon). Note: this is what we are currently doing right now with Jenkins GKE jobs (we previously had pinned versions but CI broke).

With all that said, if our release cycle is around 2-3 months anyway, updating the workflows at this time could also work (just like we do for Jenkins)...

If we do decide to go forward with the need to pin version on all providers, I would suggest we leave this out of this PR, as there is a bit of work required in order to properly pin versions on all providers without having the CI breaking all the time.

AKS: https://docs.microsoft.com/en-us/azure/aks/supported-kubernetes-versions
EKS: https://docs.aws.amazon.com/eks/latest/userguide/kubernetes-versions.html

Reply:
Don't worry, we will be monitoring updates when a new version is out.

Reply:
I see `ready-to-merge`, but was this discussion resolved?

Reply:
Yes: we'll tackle K8s version pinning at a later point, after we have wrapped up priorities for 1.10 CI 3.0 items. I have added it to the ever growing list of workflow nice to haves :D

Reply:
For the record, where is that list kept?

Reply:
For now, in our private roadmap https://github.com/isovalent/roadmap/issues/2